### ISSUE TITLE
RaspberryMatic add-in for Home Assistant returns wrong device name after reboot.

### ISSUE THREAD

**Issue by sotatech** [jens-maus/RaspberryMatic#1391]
**Describe the bug**
I have posted about this problem on Home Assistant forum here: https://community.home-assistant.io/t/homematic-keypress-giving-inconsistent-results/327602 but I didn't find a solution. I suspect it is a HA problem, but I haven't found a solution, so I wanted to check here in case it is a RaspberryMatic bug.
RaspberryMatic add-in for Home Assistant 2021.8.8 is running on HA OS with a HB-RF-ETH adapter. I have an automation on the CCU to press a virtual key every 60 seconds. This is monitored by an automation on HA to confirm that it is connected, or else it calls homematic.reconnect service. Normally the event returns "name": "HM-RCV-50 BidCoS-RF" which is correct. If I reboot the host server that HA OS runs on, it returns "name": "BidCoS-RF" causing the automation to fail. Restarting Home Assistant (not the OS) fixes the problem. Restarting RaspberrMatic add-in does not fix the problem.

**Steps to reproduce the behavior**
1. Setup an automation on the CCU to press a virtual key at a regular interval.
2. In Home Assistant - Developer Tools - Events, subscribe to homematic.keypress
3. When the event fires, note that the name is "HM-RCV-50 BidCoS-RF"
4. In Home Assistant - Supervisor - System, click Reboot Host.
5. After reboot, repeat steps 2 & 3 and note that the name is now "BidCoS-RF"
6. Restart Home Assistant to restore proper operation

**Expected behavior**
The name parameter should not change

**Screenshots**
...

**System information:**
 - RaspberryMatic Version: 3.59.6.20210807 add-on for Home Assistant
 - Used Hardware: Intel i5 NUC
 - Used HomeMatic RF-Module: HB-RF-ETH
 - Home Assistant OS 6.2

**Additional context**
...

**Comment by jens-maus** [comment]
As HA is using pyhomematic (https://github.com/danielperna84/pyhomematic) for the HA<>CCU communication perhaps @danielperna84 can comment/investigate on that matter because if a restart of the HA addon does not solve this issue it seems pyhomematic might return the name of the virtual key device somehow wrong.

However, you could also try to rename that device in the CCU/RaspberryMatic and then see if this might make a difference.

**Comment by sotatech** [comment]
Hi Jens, Daniel did comment on my HA post and asked me to try disabling the resolvenames option, but it didn't fix the problem. I will try renaming the device in the CCU and report back, but it may take a while as my family get cross when I start change things!
I can certainly work around it by testing for both names in my automation, but it would be nice to find the solution in case it causes problems for other users.

**Comment by danielperna84** [comment]
I don't know what's the reason for this problem. My first guess it that it's somehow related to [this](https://github.com/home-assistant/core/blob/5aa6f9dbb2403eadb77daefc2cefdc45712fa18b/homeassistant/components/homematic/__init__.py#L608) piece of code.

In pyhomematic the device is generally made available [here](https://github.com/danielperna84/pyhomematic/blob/361caeb47f28056a6bd7f866f83b5a3d8b2710e5/pyhomematic/devicetypes/misc.py#L91), depending on the name `HM-RCV-50`.

The code where the names are resolved in pyhomematic is [here](https://github.com/danielperna84/pyhomematic/blob/361caeb47f28056a6bd7f866f83b5a3d8b2710e5/pyhomematic/_hm.py#L459). 

I don't have the capacity to work on this in the short term. Somehow it's looking as if the JSON-RPC API is returning inconsistent results, or the result is handled incorrectly somewhere.

**Comment by sotatech** [comment]
That's OK @danielperna84 , I just wanted to document the problem in case there was a simple solution. @jens-maus I tried your suggestion of renaming the virtual device and restarting the server. The result was the same, it returned "name": "BidCoS-RF" despite the button being named "Remote Control Button".  Strangely, restarting HA reverted the name to "HM-RCV-50 BidCoS-RF" not to it's new name.
However, I just changed the name of one channel not the name of it's host device.
